### PR TITLE
Use StPharoApplication instead of custom settings application

### PR DESCRIPTION
**Pull request by hernanmd**
Includes the changes from the add_search_feature PR #873

### CODE REVIEW

**Comment by jecisc** [comment]
Can you merge development into this branch to see only the changes of this PR and not the other one please?

**Comment by hernanmd** [comment]
> Can you merge development into this branch to see only the changes of this PR and not the other one please?

Done

File: src/NewTools-SettingsBrowser-Tests/StSettingsBrowserTest.class.st [hunk 1]
```diff
@@ -13,9 +13,8 @@ Class {
 { #category : 'running' }
 StSettingsBrowserTest >> application [
 
-	^ StSettingsApplication new
-		  useBackend: #Morphic;
-		  yourself
+	^ StPharoApplication current
```
Comment (jecisc) [inline src/NewTools-SettingsBrowser-Tests/StSettingsBrowserTest.class.st]: I am really not sure, but shouldn't we do something like `StSettingBrowser new application`? Like this, if the app changes one day, it'll still be the right one? (As I said, I'm really not sure here)

**Review by jecisc — APPROVED** [review]

**Comment by jecisc** [comment]
✗ #testExternalIDEDependencies (73ms)
TestFailure: Given Collections do not match!
	additions : #('NewTools-FileBrowser')
	missing: #()

**Comment by hernanmd** [comment]
> ✗ #testExternalIDEDependencies (73ms) TestFailure: Given Collections do not match! additions : #('NewTools-FileBrowser') missing: #()

Is this failure related to this PR? 

I don't see how it could be since I didn't add FileBrowser class references.

**Comment by jecisc** [comment]
It indeed comes from the current Pharo build :(

**Comment by hernanmd** [comment]
Thanks

**Pull request merged**